### DISCUSSION TITLE
Closing Neo-tree buffer with PersistedSavePre

### DISCUSSION THREAD

**Discussion by mgokbulut** [olimorris/persisted.nvim#86] [Q&A]
First of all, thanks for making such a great plugin publicly available. I have been using persistence.nvim for a long while before I discovered your project. I would like to say it has much more to offer than the competition. My greatest reason for switching is that your project allows you to control which buffers should and shouldn't be saved. I unfortunately use a file tree while I am developing my projects and sometimes I close neovim without closing the file tree. Previously, this caused the persistence.nvim to restart the session from the file tree instead of the buffer I was currently on (which was extremely annoying for the longest time). It seems like your plugin allows me to close the file tree so that the session doesn't store that information using an Autocommand with PersistedSavePre. However what I have experienced is that if you close neovim with file tree on (I am using specifically neo-tree), It doesn't start from the buffer that you were on before closing, it starts from the buffer you were on before opening the file tree. I am not sure if closing the file tree pre-save skews the index in which you store the buffers or if it's my own stupidity configuring the plugin. I would really appreciate it if you can give me some advice on why this might be happening. Thank you for your time in advance

```
{
"olimorris/persisted.nvim",
config = true,
opts = {
	silent = true,
	autosave = true,
	autoload = true,
},
},
```
```
vim.api.nvim_create_autocmd("User", {
	pattern = "PersistedSavePre",
	group = group,
	callback = function()
		pcall(vim.cmd, "bw neo-tree")
	end,
})
```

**Comment by olimorris** [comment]
Hey @mgokbulut and thanks for your kind words.

What about if you use the `require("neo-tree.command").execute({ action = "close" })` command instead? `:bw` is always a last resort (I should update the docs and remove that example).
